db/db.go

```diff
@@ -270,6 +270,105 @@ func (db *DB) Shutdown() error {
 	return nil
 }
 
+// MockAuthDB mocks the AuthDB interface. //
+type MockAuthDB struct {
```

Comment:
You cannot export to another package things in a `*_test.go`, this mock is probably used in a different package.

Reply:
Can we just put the mock in the other package that needs it then? Or is it normal in go-land to ship mocks and stubs and whatnot with your normal code? Does go automatically strip unused code?

Reply:
The compiler is able to remove unused code. Regarding if it's normal or not, ... I don't think there's a consistent behavior, some packages my create a different package with the mocks, other's don't.

Reply:
This mock is being used by multiple different packages, because the db is used in multiple different packages.

If you guys have an opinion on where this sort of thing should live, I'm open to it. But, beyond copy pasta (which I'd prefer not to do), I don't see a way for this code to not be in the public facing code.

Reply:
Putting test stuff in public code is new to me so that's why I commented. I don't know what is typical in go, though.

Reply:
Hmm maybe for now we could just rename it. It it's a wrapper of the authdb interface. A wrapper can be useful in other contexts beyond mocking...

Reply:
Alternatively maybe we could create a test_helpers package that contains shared test code

Reply:
I'm fine with this, although at this point I'd prefer that sorta change to be part of a future commit just so we can not add more to this one.